### PR TITLE
feat(app-switcher): add other products support

### PR DESCRIPTION
**Pull request by emoralesb05**
Add support for `otherProducts` in the app switcher representing them as an expansion panel that only appears if `otherProducts` has a value/length.

Initial representation

image

After expanding 

image

Fallback

image

### CODE REVIEW

**Comment by adamnel** [comment]
**Styling stuff**
- Vantage Launcher text should use "input" text style. Looks like body-2 maybe currently.
- Between dividers and list items there should be an 8px gap (not needed on top of the first one)
- List items should use "avatar" style list items - basically they add the 40px circle behind the icons. This'll let us add colored icons more nicely to match what's going onto the full Launcher page.
- Remove divider underneath opened expansion panel.

**Structural stuff**
- Original plan for the "explore more products" area was for it to be an expansion panel. See updated designs below showing how to use it in that circumstance.
- Remove "app center" link, that's what the "view all" link will take you to.

**Updated the design to account for the more products scenario**
image

**Comment by emoralesb05** [comment]
Styling stuff

- [x] Vantage Launcher text should use "input" text style. Looks like body-2 maybe currently.

Done~

- [x] Between dividers and list items there should be an 8px gap (not needed on top of the first one)

Why is that? Im trying to avoid adding custom styles if its not part of material or our platform.

- [ ] List items should use "avatar" style list items - basically they add the 40px circle behind the icons. This'll let us add colored icons more nicely to match what's going onto the full Launcher page.

So you want us to use `lists`? Right now we are using menu items.


- [x] Remove divider underneath opened expansion panel.

Done~

Structural stuff

- [ ] Original plan for the "explore more products" area was for it to be an expansion panel. See updated designs below showing how to use it in that circumstance.

Yes, i understand that the expansion could replace the link, im 100% good with removing it. Just conversations in general applied to where the link would take the user to a place with all our prods to create awareness.. it doesnt have to be in the switcher tho. (Also, sometimes there wont be other products, so its dynamic)

- [ ] Remove "app center" link, that's what the "view all" link will take you to.

Why would it take you to AppCenter?

**Comment by adamnel** [comment]
>     * [ ]  Between dividers and list items there should be an 8px gap (not needed on top of the first one)
> 
> 
> Why is that? Im trying to avoid adding custom styles if its not part of material or our platform.
> 
> Or which divider? The one between launcher and the list?

It is a standard material thing. It also applies at the top and bottom of menus where a menu item buts up against the card edge. Trying to find where it says it explicitly but you can see it in the specs at the bottom of this page on menus, for example https://material.io/archive/guidelines/components/menus.html#menus-specs

Here's the redlines from that page showing the gaps:
image

Here's where we need it:
image

>     * [ ]  List items should use "avatar" style list items - basically they add the 40px circle behind the icons. This'll let us add colored icons more nicely to match what's going onto the full Launcher page.
> 
> 
> So you want us to use `lists`? Right now we are using menu items.

However you want to do it - not trying to reinvent anything here. Just want to match our display of apps in various places to give them that slightly larger avatar style icon.

>     * [ ]  Remove "app center" link, that's what the "view all" link will take you to.
> 
> 
> Why would it take you to AppCenter?

Where else would it take you? That's where you'd go to see as broad a selection of apps as are available to you right?

**Comment by emoralesb05** [comment]
Most changes are addressed 

image

Im not gonna change the view all because AppCenter is out of the question (as a UI) for the foreseeable future in that sense.

**Comment by adamnel** [comment]
> Most changes are addressed
> 
> image
> 
> Im not gonna change the view all because AppCenter is out of the question (as a UI) for the foreseeable future in that sense.

Fair enough. Couple remaining tweaks style-wise:
image

**Comment by emoralesb05** [comment]
Oh wow, my contrast in my screen didnt let me see the shadow. Ill fix that.

**Comment by adamnel** [comment]
OOPS meant to point that "proper noun" arrow at the "Vantage Applications" line, not "Vantage Launcher"

So, should be:
- Vantage applications
- Vantage Launcher
- Other products

**Comment by kyleledbetter** [comment]
@emoralesb05 looking good, was just peaking in the code and wanted to ask

1. Will we be able to add custom colors to each icon/avatar?
2. Will the icons be able to switch between `matListIcon` and `matListAvatar`? if so, independently for each list item?
3. Will the icons be able to switch between the default font icon `<mat-icon>apps</mat-icon>' and SVG icons `<mat-icon svgIcon="assets:blah"></mat-icon>'
4. Is this factoring in other fontSets for the icons?

pic for reference
image

**Comment by christianmemije** [comment]
Clicking on the header that says `Vantage Applications` closes the menu

**Comment by emoralesb05** [comment]
@kyleledbetter 

Will we be able to add custom colors to each icon/avatar?
Yes, its in a different PR a way to add classes to the icons

Will the icons be able to switch between matListIcon and matListAvatar? if so, independently for each list item?

No, do you want to?

Will the icons be able to switch between the default font icon <mat-icon>apps</mat-icon>' and SVG icons '

Yes, it already does that

Is this factoring in other fontSets for the icons?

It doesnt take fontSets into account, i was assuming we would largely use `svgs`. Should it?

**Comment by emoralesb05** [comment]
@christianmemije thats how mat-menu works, we would need to prevent clicks on the elements to avoid that.

**Comment by christianmemije** [comment]
Yeah we can just use `_blockEvent`

**Comment by emoralesb05** [comment]
I'll address that in a separate PR.

File: src/lib/app-switcher/app-switcher.component.html [hunk 1]
```diff
@@ -13,23 +13,48 @@
...
     </div>
     <!--content-->
-    <ng-container *ngFor="let product of productList; last as isLast">
-      <a mat-menu-item [href]="product.href">
-        <mat-icon *ngIf="product.svgIcon; else iconTemplate" [svgIcon]="product.svgIcon"></mat-icon>
-        <ng-template #iconTemplate>
-          <mat-icon>{{ product.icon || 'apps' }}</mat-icon>
-        </ng-template>
-        {{ product.text }}
+    <mat-list>
```
Comment (christianmemije) [inline src/lib/app-switcher/app-switcher.component.html]: Let's make these 2 lists nav lists

**Comment by christianmemije** [comment]
Could also be addressed in a diff PR, but the menu needs to focus on the first item when opened and be able to move around using tab.

**Comment by emoralesb05** [comment]
Yes @christianmemije , lets tackle a11y in another PR. 

I addressed the click closing issue in this one

**Pull request merged**